pftp/client_handler.go

```diff
@@ -84,119 +108,160 @@ func (c *clientHandler) setClientDeadLine(t int) {
 }
 
 func (c *clientHandler) handleCommands() error {
-	proxyError := make(chan error)
-	clientError := make(chan error)
+	// Check max client. If exceeded, send 530 error to client and disconnect
+	if atomic.LoadInt32(c.currentConnection) >= c.config.MaxConnections {
+		err := fmt.Errorf("exceeded client connection limit")
+		r := result{
+			code: 530,
+			msg:  "max client exceeded",
+			err:  err,
+			log:  c.log,
+		}
+		if err := r.Response(c); err != nil {
+			c.log.err("cannot send response to client")
+		}
+
+		c.Close()
```

Comment:
Why don't you check return value `c.Close()`?

Reply:
I thought it just close connection(if already closed, try close again).

So if check close function's return, there will return `use of closed connection`.

But now I understand what you said.

If another error occurs and the connection can not be closed, it means goroutine's leak.

So I will added to check Close function's return value and write it to log.

Reply:
With this feedback(https://github.com/pyama86/pftp/pull/32#pullrequestreview-242073405),
I fixed close function for check return value for net.Conn.Close().

If error reason is already closed connection, ignore it.
The other error will write to log.

Reply:
changed : 
https://github.com/pyama86/pftp/pull/32/commits/93d17eebf8219dc0697e3dabba65787adf303482

Reply:
Great!
I guess we could make a dedicated method.

```golang
# just idea
type closer interface {
  Close error
}
const AleadyCloseMessage = "use of closed"
func CloseConnection(c closer) error {
  if err := c.Close(); err != nil {
   if !strings.Contains(err.Error(), AleadyCloseMessage) {
     c.log.err("client connection close error: %s", err.Error())
   }
}
```

Reply:
Great idea!

I modified each structures close methods to using interface type.

https://github.com/pyama86/pftp/pull/32/commits/ad0f25120c2666368ad8d6a7feb2f8bb66639ab1